kordac/processors/GenericContainerBlockProcessor.py

```diff
@@ -0,0 +1,106 @@
+from markdown.blockprocessors import BlockProcessor
+from kordac.processors.errors.TagNotMatchedError import TagNotMatchedError
+from kordac.processors.utils import etree, parse_arguments, process_parameters, blocks_to_string
+import re
+
+
+class GenericContainerBlockProcessor(BlockProcessor):
+    def __init__(self, processor, ext, *args, **kwargs):
+        '''
+        Args:
+            ext: An instance of the Kordac Extension.
+        '''
+        super().__init__(*args, **kwargs)
+        self.processor = processor
+        self.p_start = re.compile(r'(^|\n) *\{{{0} ?(?P<args>[^\}}]*)(?<! end)\}} *(\n|$)'.format(self.processor))
+        self.p_end = re.compile(r'(^|\n) *\{{{0} end\}} *(\n|$)'.format(self.processor))
+        self.arguments = ext.processor_info[self.processor]['arguments']
+        template_name = ext.processor_info.get('template_name', self.processor)
+        self.template = ext.jinja_templates[template_name]
+        self.template_parameters = ext.processor_info[self.processor].get('template_parameters', None)
+        self.process_parameters = lambda processor, parameters, argument_values: \
+            process_parameters(ext, processor, parameters, argument_values)
+
+    def test(self, parent, block):
+        ''' Tests a block to see if the run method should be applied.
+
+        Args:
+            parent: The parent node of the element tree that children
+            will reside in.
+            block: The block to be tested.
+
+        Returns:
+            True if there are any start or end tags within the block.
+        '''
+        return self.p_start.search(block) is not None or self.p_end.search(block) is not None
+
+    def run(self, parent, blocks):
+        ''' Generic run method for container tags.
+
+        Args:
+            parent: The parent node of the element tree that children
+            will reside in.
+            blocks: A list of strings of the document, where the
+            first block tests true.
```

Comment:
Comment is a preprocessor, which is a different type of processor that takes in lines and returns those lines back with any modifications applied.